examples/ble_scanning.rs

```diff
@@ -17,31 +16,23 @@ struct LedCommand {
 
 #[libtock::main]
 async fn main() -> TockResult<()> {
-    let Drivers {
-        led_driver_factory,
-        mut ble_scanning_driver,
-        ..
-    } = libtock::retrieve_drivers()?;
+    let mut drivers = libtock::retrieve_drivers()?;
 
-    let led_driver = led_driver_factory.create_driver()?;
+    let leds_driver = drivers.leds.init_driver()?;
 
     let mut shared_buffer = BleScanningDriver::create_scan_buffer();
     let mut my_buffer = BleScanningDriver::create_scan_buffer();
-    let shared_memory = ble_scanning_driver.share_memory(&mut shared_buffer)?;
+    let shared_memory = drivers.ble_scanning.share_memory(&mut shared_buffer)?;
 
     let mut callback = BleCallback::new(|_: usize, _: usize| {
         shared_memory.read_bytes(&mut my_buffer[..]);
         ble_parser::find(&my_buffer, simple_ble::gap_data::SERVICE_DATA as u8)
             .and_then(|service_data| ble_parser::extract_for_service([91, 79], service_data))
             .and_then(|payload| corepack::from_bytes::<LedCommand>(&payload).ok())
-            .and_then(|msg| {
-                led_driver
-                    .get(msg.nr as usize)
-                    .map(|led| led.set_state(msg.st))
-            });
+            .and_then(|msg| leds_driver.get(msg.nr as usize).map(|led| led.set(msg.st)));
     });
 
-    let _subscription = ble_scanning_driver.start(&mut callback)?;
+    let _subscription = drivers.ble_scanning.start(&mut callback)?;
```

Comment:
we should not trick `#[must_use]` here or anywhere else but really drop the subscription explicitly.

Reply:
This actually produces dead code in the relevant cases so there is currently no fix.

Reply:
Calling `drop` issues a warning because of the endless `loop { ... }`